tutoraspects/plugin.py

```diff
@@ -683,111 +687,114 @@ def _mount_superset_compose(
-                RenderWidget: CourseOutlineSidebar,
-            },
-          }""",
-        ),
-        (
-            "authoring",
-            "org.openedx.frontend.authoring.course_outline_sidebar.v1",
-            """
-          {
-            op: PLUGIN_OPERATIONS.Wrap,
-            widgetId: 'default_contents',
-            wrapper: SidebarToggleWrapper,
-          }""",
-        ),
-        (
-            "authoring",
-            "org.openedx.frontend.authoring.course_unit_sidebar.v2",
-            """
-          {
-            op: PLUGIN_OPERATIONS.Insert,
-            widget: {
-                id: 'course-unit-sidebar',
-                priority: 1,
-                type: DIRECT_PLUGIN,
-                RenderWidget: UnitPageSidebar,
-            },
-          }""",
-        ),
-        (
-            "authoring",
-            "org.openedx.frontend.authoring.course_unit_sidebar.v2",
-            """
-          {
-            op: PLUGIN_OPERATIONS.Wrap,
-            widgetId: 'default_contents',
-            wrapper: SidebarToggleWrapper,
-          }""",
-        ),
-        (
-            "authoring",
-            "org.openedx.frontend.authoring.course_unit_header_actions.v1",
-            """
-          {
-              op: PLUGIN_OPERATIONS.Insert,
-              widget: {
-                  id: 'unit-header-aspects-button',
-                  priority: 60,
-                  type: DIRECT_PLUGIN,
-                  RenderWidget: CourseHeaderButton,
-              },
-          }""",
-        ),
-        (
-            "authoring",
-            "org.openedx.frontend.authoring.course_outline_header_actions.v1",
-            """
-          {
-              op: PLUGIN_OPERATIONS.Insert,
-              widget: {
-                  id: 'outline-header-aspects-button',
-                  priority: 60,
-                  type: DIRECT_PLUGIN,
-                  RenderWidget: CourseHeaderButton,
-              },
-          }""",
-        ),
-        (
-            "authoring",
-            "org.openedx.frontend.authoring.course_outline_unit_card_extra_actions.v1",
-            """
-          {
-            op: PLUGIN_OPERATIONS.Insert,
-            widget: {
-                id: 'units-action-aspects-button',
-                priority: 60,
-                type: DIRECT_PLUGIN,
-                RenderWidget: UnitActionsButton,
-            },
-          }""",
-        ),
-        (
-            "authoring",
-            "org.openedx.frontend.authoring.course_outline_subsection_card_extra_actions.v1",
-            """
-          {
-            op: PLUGIN_OPERATIONS.Insert,
-            widget: {
-                id: 'units-action-aspects-button',
-                priority: 60,
-                type: DIRECT_PLUGIN,
-                RenderWidget: SubSectionAnalyticsButton,
-            },
-          }""",
-        ),
-    ]
-)
+# If PLUGIN_SLOTS doesn't exist, we are on Redwood and do not
+# support in-context metrics.
```

Comment:
and tutor-mfe version needed to support in-context metric is v19 and above.

Reply:
@bmtcril  Let me try to understand what we are trying to do. We will create a toggle in platform-plugin-aspects, which opens up a toggle in-context API that determines if in-context metrics will be displayed or not.

Why do we need this?

The feature is still toggleable by ASPECTS_ENABLE_STUDIO_IN_CONTEXT_METRICS, but if you want to do it at the env.config.js level or  PLUGIN_SLOTS level, maybe we can add a gate here, or we can eliminate the use of PLUGIN_SLOTS.add_item by writing directly to https://github.com/openedx/tutor-contrib-aspects/blob/643aeec1e56c3fc3e0851f34740b241de07ad448/tutoraspects/patches/mfe-env-config-runtime-definitions-authoring

But these will be done at build time and not at runtime.

Let me know if I am not able to understand something.

Reply:
Now that I think more about it, making this a runtime toggle solution doesn't seem to be straightforward.  The decision needs to be taken in env,config.js or inside the plugin slots which doesn't seem to be a good solution.

Reply:
@bmtcril  sounds right :) thank you for the amazing work.